test/libyul/YulOptimizerTestCommon.cpp

```diff
@@ -194,6 +194,8 @@ YulOptimizerTestCommon::YulOptimizerTestCommon(
 		}},
 		{"fullSimplify", [&]() {
 			disambiguate();
+			FunctionGrouper::run(*m_context, *m_ast);
+			BlockFlattener::run(*m_context, *m_ast);
```

Comment:
It's odd that this change does not affect any tests.

Take a look e.g. at https://github.com/ethereum/solidity/blob/38a5852128f14fe84589e18f655da8af23c1b6fb/test/libyul/yulOptimizerTests/fullSimplify/not_applied_function_call_equality_not_movable.yul#L10-L14

The function does not look grouped (it's in the middle of other instructions) and there's an extra unflattened block.

Reply:
Updated PR with the requested change. Please note that the simplified test in the first commit of this PR fails, so I have added the Yul code after failed optimization as expectation. In the second commit, the fix is added and test updated. Hope this helps!